contracts/StableJoeStaking.sol

```diff
@@ -125,7 +133,11 @@ contract StableJoeStaking is Initializable, OwnableUpgradeable {
     function deposit(uint256 _amount) external {
         UserInfo storage user = userInfo[_msgSender()];
 
-        uint256 _fee = _amount.mul(depositFeePercent).div(DEPOSIT_FEE_PERCENT_PRECISION);
+        uint256 _fee;
+        // Only EOAs holding Smol Joes are exempt from paying the deposit fee
+        if (smolJoes.balanceOf(_msgSender()) == 0 || _msgSender() != tx.origin) {
```

Comment:
This will revert on `smolJoes` being address 0, (this will be the case during upgrade)

Reply:
I believe is talking about the case between when the contract is upgraded and the `initializeSmalJoes` function is called (which may be a very small amount of time). You could do this in a backwards compatible way by changing this to 
`if (address(smolJoes) != 0 && smolJoes.balanceOf(_msgSender()) == 0 || _msgSender() != tx.origin)` this works due to the [short circuiting](https://docs.soliditylang.org/en/latest/types.html#booleans) behaviour of `&&` and `||`